DogsList/DogsList/Controllers/SearchHistoryViewController.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  SearchHistoryViewController.swift
+//  DogsList
+//
+//  Created by Nick Krasnitskiy on 16.07.2021.
+//
+
+import UIKit
+
+class SearchHistoryViewController: UIViewController {
+    
+    enum Section: CaseIterable {
+        case main
+    }
+    
+    let searchHistory: [String]
+    
+    init(with searchHistory: [String]) {
+        self.searchHistory = searchHistory
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    typealias DataSource = UICollectionViewDiffableDataSource<Section, String>
+    private var dataSource: DataSource!
+    private var collectionView: UICollectionView! = nil
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureCollectionView()
+        configureDataSource()
+        addSnaphot(searchHistory: searchHistory.removingDuplicates())
+        navigationItem.title = "Search History"
+        navigationController?.navigationBar.tintColor = .white
+    }
+}
+
+extension SearchHistoryViewController {
+    
+    private func createLayout() -> UICollectionViewLayout {
```

Comment:
You can group all private methods under a private extension. 

```
private extension SearchHistoryViewController {
    func createLayout() -> UICollectionViewLayout {
    ...
    }
    ...
}
```

Reply:
Ok. Fixed.